lib/util/timestamp.js

```diff
@@ -1,16 +1,84 @@
-// Get a date object in the correct format, without requiring a full out library
-// like "moment.js".
-function yyyymmddhhmmss() {
-  const d = new Date();
-
-  return (
-    d.getFullYear().toString() +
-    (d.getMonth() + 1).toString().padStart(2, '0') +
-    d.getDate().toString().padStart(2, '0') +
-    d.getHours().toString().padStart(2, '0') +
-    d.getMinutes().toString().padStart(2, '0') +
-    d.getSeconds().toString().padStart(2, '0')
-  );
+function yyyymmddhhmmss(...time) {
```

Comment:
What's the point of adding this parameter if it's never passed from anywhere? Also what's the benefit of having it over just using UTC always?

Reply:
>  Also what's the benefit of having it over just using UTC always?

I'm wondering the same.

Reply:
the point of having the parameter is that if a team doesnt want UTC, lets say they are all co-located in a EST timezone, then this parameter would allow for, in the future, letting that timezone be given somewhere else. For instance, if a team wanted to all use EST, possibly if a parameter was added into the knexfile as timezone: EST, then the function could translate the offset for that if it is given in as a parameter. That sections definitely needs a bit of work depending on how that could/would get implemented in the future, I just thought teams should be given the choice of using the default of UTC or changing to their own specified timezone.

Reply:
@Tolterix But it's never passed from knexfile into `yyyymmddhhmmss` right now.
Still, why any team might want more specific time, considering that there is no significance to the timestamp whatsoever, and it is used merely to ensure that seed/migration files are automatically sorted by the order of their creation, which is achieved with UTC perfectly?

Reply:
The fact that it is never passed in now won't affect it at all, as I am using a Rest API with the ...time. The functionality of how you use it everywhere in the code right now wouldn't change other than changing the time to UTC and adding the string 'UTC' to the beginning of the timestamp, if you implement that part of this. The main part about the sorting and creation falls back to when things are migrated in teams. I have discussed this with a couple of my coworkers who have worked with this and they have concluded that having their own timezone would help for clarity for them dealing with migrations. @kibertoad

Reply:
@Tolterix What would be the clarity about? Why would anyone care at what particular hour seed file was created?

Reply:
Not the seed, but definitely when migrations happen.

Reply:
I guess that tracking when migrations are made would help when doing multiple migrations that are very similar if you don't delete them. Might be an edge case I will admit, and UTC time as default for everyone would definitely resolve the base issue. Other timezones was just a nicety that I thought would be good to have in the future.

Reply:
I still don't see any reason for not using UTC... commit history tells when migration was created and necessarily migrations are not even applied in the same order if some feature whose migration was created later on is merged to the production first etc. Also nothing prevents your team to add any timezone or any other tag to the migration name even if it is not generated automatically. 

Why is it necessary to know in what exact time in some exact timezone certain migration file has been created? (it is not even exactly the same time when that migration contents has been written...).

Reply:
how does migrate latest work if it isnt looking at that timestamp? or is the timestamp literally just a randomly generated id and it doesnt matter what it is? because if that is the case you could just remove the timestamp function and put 20 random characters on the front of the files name.

Reply:
@Tolterix `latest` retrieves list of migrations that were already run, compares it against full list of migrations, and runs everything that wasn't run yet. As I've said, timestamp is used to ensure that migrations are sorted by creation date, and for that purpose always using UTC is perfectly appropriate.

Reply:
Awesome, yeah with that knowledge it makes perfect sense to refactor it to not include all the other stuff and just use UTC. I'll refactor my repo appropriately and do another pull after.

Reply:
Thank you!